Lib/difflib.py

```diff
@@ -1266,6 +1266,8 @@ def _check_types(a, b, *args):
     if b and not isinstance(b[0], str):
         raise TypeError('lines to compare must be str, not %s (%r)' %
                         (type(b[0]).__name__, b[0]))
+    if isinstance(a, str) or isinstance(b, str):
```

Comment:
Following the existing code, IMO it would be great add the `not %s (%r)'` phrase.